src/mark.typ

```diff
@@ -289,5 +296,12 @@
       snap-to: snap-to)
   }
 
+  // If not transformed pre mark placement,
+  // transform everything after mark placement.
+  if not is-transformed {
+    segments = drawable.apply-transform-segments(transform, segments)
+    drawables = drawable.apply-transform(transform, drawables)
+  }
```

Comment:
As said in `draw/shapes.typ` if this pushes the mark onto the path drawables, you would then only need a single `apply-transform` call and no `apply-transform-segments` call.
You would also be able to just use `apply-transform` on line 264.